libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,257 @@
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
-    assertEquals(4, navigationWithOptions.getMilestones().size());
+
     navigationWithOptions.removeMilestone(null);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_correctMilestoneWithIdentifierGetsRemoved() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
-    Milestone milestone = new StepMilestone.Builder().setIdentifier(5678).build();
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+    int removedMilestoneIdentifier = 5678;
+    Milestone milestone = new StepMilestone.Builder().setIdentifier(removedMilestoneIdentifier).build();
     navigationWithOptions.addMilestone(milestone);
-    assertEquals(1, navigationWithOptions.getMilestones().size());
-    navigationWithOptions.removeMilestone(5678);
+
+    navigationWithOptions.removeMilestone(removedMilestoneIdentifier);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_noMilestoneWithIdentifierFound() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
-    assertEquals(1, navigationWithOptions.getMilestones().size());
-    navigationWithOptions.removeMilestone(5678);
+    int removedMilestoneIdentifier = 5678;
+
+    navigationWithOptions.removeMilestone(removedMilestoneIdentifier);
+
     assertEquals(1, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void getLocationEngine_returnsCorrectLocationEngine() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
     LocationEngine locationEngine = mock(LocationEngine.class);
-    LocationEngine locationEngine2 = mock(LocationEngine.class);
+    LocationEngine locationEngineInstanceNotUsed = mock(LocationEngine.class);
+
     navigation.setLocationEngine(locationEngine);
```

Comment:
Should be in the `Arrange` part 👀 `getLocationEngine_returnsCorrectLocationEngine` is telling us what's the `Act` here `getLocationEngine`.